ARROW-4180: [Java] Make CI tests use logback.xml

pravindra opened the pull request:
- Add a logback.xml for each project with the default log level set to info.
- Set the "verbose" property for git-commit-plugin to false
- quieten output from qos.logback by attaching the NopStatusListener
- Fix a bunch of doc errors in ArrowBuf.java

BryanCutler commented on ci/travis_script_java.sh:
```diff
@@ -27,7 +27,7 @@ if [ $ARROW_TRAVIS_JAVA_BUILD_ONLY == "1" ]; then
     mvn -DskipTests=true -Dcheckstyle.skip=true -B install
 else
-    mvn -B install
+    mvn -B install -Dlogback.configurationFile=file:${ARROW_HOME}/java/dev/logback.xml
```
It doesn't look like it's able to find the logback file. You might try `$JAVA_DIR/dev/logback.xml`

BryanCutler commented:
Also another source of a lot of output is the `git-commit-id-plugin`. If we can just set verbose to false, that cuts down the output by about 1/3 for me.

fsaintjacques commented:
Why do we need to explicit the logback configuration in maven invocation? Can we just increase the default log level to NOTICE/INFO, and those who wants debug overwrite?

pravindra commented:
> Why do we need to explicit the logback configuration in maven invocation? Can we just increase the default log level to NOTICE/INFO, and those who wants debug overwrite?

In the PR, I haven't changed the maven invocation. I instead added a logback.xml in the corresponding directories with log level at WARN, which is picked up automatically by the tests.

fsaintjacques commented:
My bad, I had the JIRA ticket in my mind! :+1: from me.

BryanCutler commented on ci/travis_script_java.sh:
```diff
@@ -22,8 +22,8 @@ set -e
 
-export MAVEN_OPTS="$MAVEN_OPTS -Dorg.slf4j.simpleLogger.defaultLogLevel=warn"
-if [ $ARROW_TRAVIS_JAVA_BUILD_ONLY == "1" ]; then
+export MAVEN_OPTS="$MAVEN_OPTS -Dorg.slf4j.simpleLogger.defaultLogLevel=info"
```
Why change the log level here?

pravindra replied:
without this, the output was too concise. We can't figure out for example if it ran "memory" tests or not. Or, how many tests ran.

https://travis-ci.org/apache/arrow/jobs/482786298

BryanCutler replied:
Yeah, it's probably best to have some output and hopefully with these changes it will be much smaller and not interfere.

How about just removing these exports then since `info` is the default maven log level anyway?

BryanCutler commented:
@pravindra could you please add a description in the PR?

pravindra commented:
> @pravindra could you please add a description in the PR?

done.

pravindra commented:
> It seems like you need to add an appender now to get show the logs in STDOUT

That was not intentional. I'll fix this.

pravindra commented:
> LGTM, I ran this and it has much less output.Would you mind updating the usage a little bit in the java/README.md here https://github.com/apache/arrow/tree/master/java#test-logging-configuration? It seems like you need to add an appender now to get show the logs in STDOUT

updated the README and added an appender by default.

BryanCutler commented:
merged to master, thanks @pravindra !
